index.bs

```diff
@@ -1739,6 +1782,212 @@ contexts that have active documents; navigation can also cause contexts to
 become inaccessible but not yet get discarded because bfcache.
 </div>
 
+## Log ## {#module-log}
+
+The <dfn export for=modules>log</dfn> module contains functionality and events
+related to logging.
+
+### Definition ### {#module-log-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+Command = (GetCommand)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+LogEvent = (
+  LogEntryAddedEvent
+)
+</pre>
+
+### Types ### {#module-log-types}
+
+#### log.LogEntry #### {#types-log-logentry}
+
+<pre class="cddl local-cddl">
+
+LogLevel = "debug" / "info" / "warning" / "error"
+
+LogEntry = {
+  type: text,
+  level: LogLevel,
+  text: text / null,
```

Comment:
Is it possible for `text` to be null here? In the "console Printer steps" it won't ever be null but rather an empty string. Maybe make this non-nullable until we find a case where null makes sense?

`console.log()` with no arguments is an interesting case here, but letting text be the empty string in this case seems OK to me. One could still distinguish it from `console.log('')` based on the arguments, if that were ever important.

Reply:
I was thinking of cases like `groupEnd()` where there isn't a meaningful log message.

Reply:
Ah, that makes sense. Will we need to patch https://console.spec.whatwg.org/#groupend to support that case?

Reply:
Yeah, the patch I have isn't quite enough.

Reply:
I changed the proposed patch to intercept anything that calls a method on the console interface.